src/test/java/org/sonatype/nexus/ci/nxrm/v3/MoveComponentsStepTest.groovy

```diff
@@ -0,0 +1,248 @@
+    and: 'ListBox has empty item'
+      listBoxModel.get(0).name == FormUtil.EMPTY_LIST_BOX_NAME
+      listBoxModel.get(0).value == FormUtil.EMPTY_LIST_BOX_VALUE
+
+    and: 'ListBox is populated'
+      listBoxModel.get(1).name == repositories.get(0).name
+      listBoxModel.get(1).value == repositories.get(0).name
+  }
+
+  def 'it successfully completes a move operation based on a tag'() {
+    setup:
+
+      def project = getProject('localhost', 'maven-releases', 'foo')
+
+    when:
+      def build = project.scheduleBuild2(0).get()
+
+    then:
+      1 * nxrm3Client.move('maven-releases', 'foo')
+      jenkinsRule.assertBuildStatus(SUCCESS, build)
+  }
+
+  def 'it fails to complete a move operation based on a tag'() {
+    setup:
+
+      def project = getProject('localhost', 'maven-releases', 'foo',
+          { throw new RepositoryManagerException("Move failed") })
+
+    when:
+      def build = project.scheduleBuild2(0).get()
+
+    then:
+      jenkinsRule.assertBuildStatus(Result.FAILURE, build)
+      jenkinsRule.assertLogContains("Move failed", build)
+  }
+
+  def 'it successfully completes a move operation based on a tag as workflow'() {
+    setup:
+
+      def project = getWorkflowProject('localhost', 'maven-releases', 'foo')
+
+    when:
+      def build = project.scheduleBuild2(0).get()
+
+    then:
+      1 * nxrm3Client.move('maven-releases', 'foo')
+      jenkinsRule.assertBuildStatus(SUCCESS, build)
+  }
+
+  def 'it fails to complete a move operation based on a tag as a workflow'() {
+    setup:
+
+      def project = getWorkflowProject('localhost', 'maven-releases', 'foo'
+          , { throw new RepositoryManagerException("Move failed") })
+
+    when:
+      def build = project.scheduleBuild2(0).get()
+
+    then:
+      jenkinsRule.assertBuildStatus(Result.FAILURE, build)
+      jenkinsRule.assertLogContains("Failing build due to: Move failed", build)
+  }
+
+  def 'it fails attempting to get an nxrm3 client with invalid id'() {
+    setup:
+      def project = getProject('invalidclient', 'maven-releases', 'foo',
+          { throw new RepositoryManagerException("localhost not found") } )
+
+    when:
+      def build = project.scheduleBuild2(0).get()
+
+    then:
+      jenkinsRule.assertBuildStatus(Result.FAILURE, build)
+      jenkinsRule.assertLogContains("localhost not found", build)
+  }
+
+  def 'it fails due to missing parameter in workflow dsl'() {
+    setup:
+
+      def instance = 'someinstance'
+      def tagName = 'someTag'
+      def config = createNxrm3Config(instance)
+
+      def project = jenkinsRule.createProject(WorkflowJob.class, "nexusStagingMove")
+      project.setDefinition(new CpsFlowDefinition("node {moveComponents nexusInstanceId: '" + instance +"', tagName: '" + tagName + "'}"))
+
+      GroovyMock(RepositoryManagerClientUtil.class, global: true)
+      RepositoryManagerClientUtil.nexus3Client(config.serverUrl, config.credentialsId) >> { new IllegalArgumentException("missing arg") }
```

Comment:
I think this test is flawed due to this line; in trying to replicate the test I accidentally copied a different line from my source that did not throw an error on the client retrieval which produces different results, but I realized when I looked over yours again that I don't think you want an error to come from that point, but rather the Jenkins instance to actually error out when the step is called with invalid args.

My latest commit added the same style of test, please check it out for comparison and let me know if I did something weird.

Reply:
@wrightkris Will take a look...at a quick glance I believe you are correct..thanks for spotting it!